ts/src/transports/Transport.ts

```diff
@@ -27,40 +26,42 @@ export interface TransportOptions {
   onEnd: (err?: Error) => void;
 }
 
-let selectedTransport: TransportConstructor;
-export function DefaultTransportFactory(transportOptions: TransportOptions): Transport | Error {
-  // The transports provided by DefaultTransportFactory do not support client-streaming
-  if (transportOptions.methodDefinition.requestStream) {
-    return new Error("No transport available for client-streaming (requestStream) method");
-  }
+export function DefaultHttpTransport(transportOptions: TransportOptions): Transport {
+  return HttpTransport({ withCredentials: false })(transportOptions);
+}
 
-  if (!selectedTransport) {
-    selectedTransport = detectTransport();
-  }
+export interface HttpTransportInit {
+  withCredentials?: boolean
+}
 
-  return selectedTransport(transportOptions);
+export interface TransportFactory {
+  (options: TransportOptions): Transport;
 }
 
-function detectTransport(): TransportConstructor {
+export function HttpTransport(init: HttpTransportInit): TransportFactory {
   if (detectFetchSupport()) {
-    return fetchRequest;
-  }
-
-  if (detectMozXHRSupport()) {
-    return mozXhrRequest;
+    return FetchReadableStreamTransport({ credentials: init.withCredentials ? 'include' : 'same-origin' })
```

Comment:
How do you configure this to `omit` now?

Reply:
This implementation allows explicitly using `transport: FetchReadableStreamTransport({ credentials: "omit" })` in the invocation/client intead of `transport: HttpTransport({ /* whatever */ })`.
`HttpTransport` can't support `omit` because it's not in the "greatest common interface", as `XMLHttpRequest` can only allow `include` or `same-origin` - https://xhr.spec.whatwg.org/#the-send()-method see *credentials mode* at (6) there.